From: Borislav Petkov <bp@suse.de>
Subject: x86/MCE: Fix kABI for new AMD bank names
Patch-mainline: never, kabi
References: bsc#1128415

Disable ksyms checking for the new enum smca_bank_types. They should not be used
by anything else anyway.

Signed-off-by: Borislav Petkov <bp@suse.de>
---
 arch/x86/include/asm/mce.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/arch/x86/include/asm/mce.h
+++ b/arch/x86/include/asm/mce.h
@@ -360,17 +360,23 @@ enum smca_bank_types {
 	SMCA_FP,	/* Floating Point */
 	SMCA_L3_CACHE,	/* L3 Cache */
 	SMCA_CS,	/* Coherent Slave */
+#ifndef __GENKSYMS__
 	SMCA_CS_V2,	/* Coherent Slave */
+#endif
 	SMCA_PIE,	/* Power, Interrupts, etc. */
 	SMCA_UMC,	/* Unified Memory Controller */
 	SMCA_PB,	/* Parameter Block */
 	SMCA_PSP,	/* Platform Security Processor */
+#ifndef __GENKSYMS__
 	SMCA_PSP_V2,	/* Platform Security Processor */
+#endif
 	SMCA_SMU,	/* System Management Unit */
+#ifndef __GENKSYMS__
 	SMCA_SMU_V2,	/* System Management Unit */
 	SMCA_MP5,	/* Microprocessor 5 Unit */
 	SMCA_NBIO,	/* Northbridge IO Unit */
 	SMCA_PCIE,	/* PCI Express Unit */
+#endif
 	N_SMCA_BANK_TYPES
 };
 
